rosetta/static/admin/rosetta/js/rosetta.js

```diff
@@ -1,157 +1,228 @@
-                var lang = rosetta_settings.MESSAGES_SOURCE_LANGUAGE_CODE + "-" + destLangRoot;
-
-                a.attr("class", "suggesting").html("...");
-
-                var apiData = {
+            suggest((orig, setTranslation, setError) => {
+                const apiUrl = "https://translate.yandex.net/api/v1.5/tr.json/translate";
+                const destLangRoot = rosetta_settings.rosetta_i18n_lang_code.split("-")[0];
+                const lang = rosetta_settings.MESSAGES_SOURCE_LANGUAGE_CODE + "-" + destLangRoot;
+                const apiData = {
                     error: "onTranslationError",
                     success: "onTranslationComplete",
                     lang: lang,
                     key: rosetta_settings.YANDEX_TRANSLATE_KEY,
                     format: "html",
                     text: orig,
                 };
-
-                $.ajax({
-                    url: apiUrl,
-                    data: apiData,
-                    dataType: "jsonp",
-                    success: function (response) {
-                        if (response.code == 200) {
-                            trans.val(
-                                response.text[0]
-                                    .replace(/<br>/g, "\n")
-                                    .replace(/<\/?code>/g, "")
-                                    .replace(/&lt;/g, "<")
-                                    .replace(/&gt;/g, ">"),
-                            );
-                            a.hide();
-                        } else {
-                            a.text(response);
-                        }
-                    },
-                    error: function (response) {
-                        a.text(response);
-                    },
+                jsonp(apiUrl, apiData, (response) => {
+                    if (response.code === 200) {
+                        setTranslation(
+                            response.text[0]
+                                .replace(/< ?br>/g, "\n")
+                                .replace(/< ?\/? ?code>/g, "")
+                                .replace(/&lt;/g, "<")
+                                .replace(/&gt;/g, ">"),
+                        );
+                    } else {
+                        setError(response);
+                    }
                 });
             });
         }
     }
 
-    $("td.plural").each(function () {
-        var td = $(this);
-        var trY = parseInt(td.closest("tr").offset().top);
-        $("textarea", $(this).closest("tr")).each(function (j) {
-            var textareaY = parseInt($(this).offset().top) - trY;
-            $($(".part", td).get(j)).css("top", textareaY + "px");
+    // For each translation field textarea
+    document.querySelectorAll(".translation textarea").forEach((textarea, i) => {
+        // Focus on the first textarea on page load
+        if (i === 0) {
+            textarea.focus();
+        }
```

Comment:
Thinking about accessibility, I guess focusing on the first textarea is probably not a good idea:

* I imagine people who use the tab key for navigation expect it to start from the beginning of the page. 
* Making an on-screen keyboard show up on page load is likely not helpful in most cases either.
* Some people use the space bar to scroll down, which makes a change in the translation instead.

So I'd rather remove this.